pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -91,49 +92,154 @@ func (ic *ifConfigurator) moveIfToNetns(ifname string, netns ns.NetNS) error {
+func (ic *ifConfigurator) configureLinkAndIPAM(link netlink.Link, containerID string, containerIfaceName string, netnsPath string, mtu int, result *current.Result) error {
 	hostIface := result.Interfaces[0]
 	containerIface := result.Interfaces[1]
 
+	err := ic.netlink.LinkSetMTU(link, mtu)
+	if err != nil {
+		return fmt.Errorf("failed to set MTU for VF netdevice %s: %w", containerIfaceName, err)
+	}
+	err = ic.netlink.LinkSetUp(link)
+	if err != nil {
+		return fmt.Errorf("failed to set link up to VF netdevice %s: %w", containerIfaceName, err)
+	}
+	containerIface.Name = containerIfaceName
+	containerIface.Mac = link.Attrs().HardwareAddr.String()
+	containerIface.Sandbox = netnsPath
+	klog.V(2).Infof("hostIface: %+v, containerIface: %+v", hostIface, containerIface)
+	klog.V(2).Infof("Configuring IP address for container %s", containerID)
+	// result.Interfaces must be set before this.
+	if err := ipamConfigureIface(containerIface.Name, result); err != nil {
+		return fmt.Errorf("failed to configure IP address for container %s: %w", containerID, err)
+	}
+	klog.V(2).Infof("ipam.ConfigureIface result: %+v, err: %v", result, err)
+	return nil
+}
+
+func (ic *ifConfigurator) moveVFtoContainerNS(vfNetDevice string, containerID string, containerNetNS string, containerIfaceName string, mtu int, result *current.Result) error {
 	// Move VF to Container namespace
 	netns, err := nsGetNS(containerNetNS)
 	if err != nil {
-		return fmt.Errorf("failed to open container netns %s: %v", containerNetNS, err)
+		return fmt.Errorf("failed to open container netns %s: %w", containerNetNS, err)
 	}
-	err = ic.moveIfToNetns(vfNetDevice, netns)
+	err = ic.moveIFToNetNS(vfNetDevice, netns)
 	if err != nil {
-		return fmt.Errorf("failed to move VF %s to container netns %s: %v", vfNetDevice, containerNetNS, err)
+		return fmt.Errorf("failed to move VF %s to container netns %s: %w", vfNetDevice, containerNetNS, err)
 	}
 	netns.Close()
 
 	if err := nsWithNetNSPath(containerNetNS, func(hostNS ns.NetNS) error {
 		err = renameInterface(vfNetDevice, containerIfaceName)
 		if err != nil {
-			return fmt.Errorf("failed to rename VF netdevice as containerIfaceName %s: %v", containerIfaceName, err)
+			return fmt.Errorf("failed to rename VF netdevice as containerIfaceName %s: %w", containerIfaceName, err)
 		}
 		link, err := ic.netlink.LinkByName(containerIfaceName)
 		if err != nil {
-			return fmt.Errorf("failed to find VF netdevice %s: %v", containerIfaceName, err)
+			return fmt.Errorf("failed to find VF netdevice %s: %w", containerIfaceName, err)
 		}
-		err = ic.netlink.LinkSetMTU(link, mtu)
+		return ic.configureLinkAndIPAM(link, containerID, containerIfaceName, containerNetNS, mtu, result)
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ic *ifConfigurator) moveVFtoContainerNSViaTempNS(vfNetDevice string, containerID string, containerNetNS string, containerIfaceName string, mtu int, result *current.Result) error {
```

Comment:
How about these two func names:

moveVFtoContainerNS: for the secondary network case.
moveOffloadVFToContainerNS: for the offload case.